classes/migration/upgrade/v3_4_0/I7513_DoiSettings.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/**
+ * @file classes/migration/upgrade/v3_4_0/I7513_DoiSettings.php
+ *
+ * Copyright (c) 2014-2022 Simon Fraser University
+ * Copyright (c) 2000-2022 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class I7513_DoiSettings
+ * @brief Database migrations for DOI settings refactor.
+ */
+
+namespace APP\migration\upgrade\v3_4_0;
+
+use Illuminate\Support\Facades\DB;
+
+class I7513_DoiSettings extends \PKP\migration\Migration
+{
+    /**
+     * @inheritDoc
+     */
+    public function up(): void
+    {
+        DB::table('plugin_settings')
+            ->where('plugin_name', '=', 'crossrefexportplugin')
+            ->where('setting_name', '=', 'automaticRegistration')
+            ->delete();
+
+        DB::table('plugin_settings')
+            ->where('plugin_name', '=', 'crossrefexportplugin')
+            ->update(['plugin_name' => 'crossrefplugin']);
+
+        DB::table('plugin_settings')
+            ->where('plugin_name', '=', 'dataciteexportplugin')
+            ->where('setting_name', '=', 'automaticRegistration')
+            ->delete();
+
+        DB::table('plugin_settings')
+            ->where('plugin_name', '=', 'dataciteexportplugin')
+            ->update(['plugin_name' => 'dataciteplugin']);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function down(): void
+    {
+        DB::table('plugin_settings')
+            ->where('plugin_name', '=', 'crossrefplugin')
+            ->whereNot('setting_name', '=', 'enabled')
+            ->update(['plugin_name' => 'crossrefexportplugin']);
+
+        DB::table('plugin_settings')
+            ->where('plugin_name', '=', 'dataciteplugin')
+            ->whereNot('setting_name', '=', 'enabled')
+            ->update(['plugin_name' => 'dataciteexportplugin']);
+    }
```

Comment:
...but if you're saving in a `$this->settings` object, don't forget that migrations can also be invoked with `lib/pkp/tools/migration.php`, with either `up` or `down` specified on the command line. In that case `$this->settings` won't be initialized. If you go that route, check for `$this->settings` to be initialized in `down()`; if it's not, throw a `DowngradeNotSupportedException`. That way both execution paths will be well supported.

Reply:
Ahh, hmm. In that case there may be a bunch of migrations that fail because of this. We might need to review all of the 3.4 migrations for such cases.

Reply:
I mentioned in a comment above, but the deletion of the `automaticRegistration` setting has already been handled at this point (in `I7014_DoiMigration`), so now the migration would only be renaming the plugin name in the settings table—meaning the down would completely restore the changes in up.

Reply:
(I've added a note in the development Mattermost channel for folks to review and fix their `down()` migrations as needed. It's pretty unlikely that people will be running `down()` migrations from the CLI, but it's good hygiene to support it.)